hawkrest/__init__.py

```diff
@@ -92,11 +111,14 @@ def authenticate(self, request):
         # Pass our receiver object to the middleware so the request header
         # doesn't need to be parsed again.
         request.META['hawk.receiver'] = receiver
-        return on_success
+        return self._user_lookup(request, receiver.resource.credentials)
 
     def authenticate_header(self, request):
         return 'Hawk'
 
+    def get_user(self, user_id):
```

Comment:
Update: it *is* called in Django's `AuthenticationMiddleware` class (see [here](https://github.com/django/django/blob/master/django/contrib/auth/middleware.py#L12), and called by `django.contrib.auth.get_user` [here](https://github.com/django/django/blob/master/django/contrib/auth/__init__.py#L213). DRF's `BaseAuthentication` class [doesn't inherit](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authentication.py#L36) any of Django's auth classes directly, but since `get_user()` is called while traversing the auth middleware list, I think having this method available increases reusability (ie you can use it outside DRF).